src/tests/ftest/container/per_server_fault_domain.py

```diff
@@ -0,0 +1,327 @@
+        stop_rank_1 = non_svc_ranks[0]
+        stop_rank_2 = None
+        for rank in range(8):
+            if rank != stop_rank_1:
+                if rank_to_ip[rank] == rank_to_ip[stop_rank_1]:
+                    stop_rank_2 = rank
+                    break
+        self.log.info("Stop rank 1 = %s; 2 = %s", stop_rank_1, stop_rank_2)
+
+        # Check if the rank found in the previous step is in the list. If so, remove it.
+        if stop_rank_2 in non_svc_ranks:
+            non_svc_ranks.remove(stop_rank_2)
+
+        # Remove the first element (stop_rank_1) from the list.
+        non_svc_ranks.remove(stop_rank_1)
+
+        # Now the list contains one or two elements. Select the first element and find the
+        # other rank that's on the same node as before.
+        stop_rank_3 = non_svc_ranks[0]
+        stop_rank_4 = None
+        for rank in range(8):
+            if rank != stop_rank_3:
+                if rank_to_ip[rank] == rank_to_ip[stop_rank_3]:
+                    stop_rank_4 = rank
+                    break
+        self.log.info("Stop rank 3 = %s; 4 = %s", stop_rank_3, stop_rank_4)
+
+        ranks_to_stop = f"{stop_rank_1},{stop_rank_2},{stop_rank_3},{stop_rank_4}"
+        self.log.info("Ranks to stop = %s", ranks_to_stop)
+
+        properties = self.params.get("rf_2", "/run/cont_property/*")
+
+        # Run step 2 to 8.
+        self.verify_per_server_fault(
+            create_pool=False, ranks_to_stop=ranks_to_stop, properties=properties,
+            expected_status="HEALTHY")
+
+    def test_rf2_unclean(self):
+        """Jira ID: DAOS-11200
+
+        Test Description:
+        1. Determine the ranks to stop. Three ranks in three nodes. Select up to two
+        service ranks.
+        2. Create a container with rf_lvl:2,rf:2
+        3. Run IOR to write some data to the container.
+        4. Stop the ranks.
+        5. Wait for rebuild to finish.
+        6. Get container's health status.
+        7. Restart the stopped servers for cleanup.
+        8. Verify container's Health status is HEALTHY.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=container
+        :avocado: tags=per_server_fault_domain,rf2_unclean
+        """
+        # 1. Determine the ranks to stop; three ranks in three nodes. We can select up to
+        # two ranks from service ranks. If we stop more than two service ranks, many of
+        # the dmg commands will hang.
+        # Create a pool to determine the service ranks.
+        self.pool = self.get_pool()
+        self.log.info("Pool service ranks = %s", self.pool.svc_ranks)
+
+        # Create the list of non-service ranks. Assume there are 5 service ranks and rank
+        # numbering is consecutive.
+        non_svc_ranks = []
+        host_count = len(self.hostlist_servers)
+        engines_per_host = self.params.get("engines_per_host", "/run/server_config/*")
+        rank_count = host_count * engines_per_host
```

Comment:
The recomended way to get this information is to use the `DaosServerManager ` class:
```suggestion
        rank_count = self.server_manager[0].engines
```